src/index.tsx

```diff
@@ -1,22 +1,41 @@
 import * as React from "react";
 import { render } from "react-dom";
 
-import { ApolloClient, ApolloProvider, useQuery, InMemoryCache } from "@apollo/client";
-import { GET_CHARACTER } from "./queries";
-import { GetCharactersQuery } from "./common/graphql/types/graphql";
+import {
+  ApolloClient,
+  ApolloProvider,
+  useQuery,
+  InMemoryCache,
+} from "@apollo/client";
+import { GET_CHARACTER, CharactersCountFragment } from "./queries";
+import { FragmentType, useFragment } from "./common/graphql/types";
 
 const client = new ApolloClient({
   uri: "https://rickandmortyapi.com/graphql",
   cache: new InMemoryCache(),
 });
 
+type CharactersCountProps = {
+  characters: FragmentType<typeof CharactersCountFragment>;
+};
+
+const CharactersCount = (props: CharactersCountProps) => {
+  const characters = useFragment(CharactersCountFragment, props.characters);
+
+  return <div>{characters?.info?.count}</div>;
+};
+
 const ShowPosts = () => {
-  const { loading, error, data } = useQuery<GetCharactersQuery>(GET_CHARACTER);
+  const { loading, error, data } = useQuery(GET_CHARACTER);
 
   if (loading) return <p>Loading...</p>;
-  if (error) return <p>Error :(</p>;
+  if (error || !data || !data.characters) return <p>Error :(</p>;
 
-  return <div key={1}>{data?.characters?.info?.count}</div>;
+  return (
+    <>
+      <CharactersCount characters={data.characters} />
```

Comment:
We need to pass `data.characters` to `<CharactersCount>` because that's where we'll use `characters` from the top-level query.